src/ripple/rpc/handlers/LedgerData.cpp

```diff
@@ -48,15 +48,16 @@ Json::Value doLedgerData (RPC::Context& context)
     if (!lpLedger)
         return jvResult;
 
+    bool const isMarker = params.isMember (jss::marker);
     boost::optional<ReadView::key_type> key = ReadView::key_type();
-    if (params.isMember (jss::marker))
+    if (isMarker)
```

Comment:
Looks like a mistake to me, good eyes @nbougalis! Also - If a marker was present and the optional null, calling SetHex would be a problem. An impossibility with key_type() as is, but could be a problem if changed.